phoebe/backend/contacts_smoothing.py

```diff
@@ -0,0 +1,241 @@
+
+    coords_fit_y1, teffs_fit_y1 = _isolate_sigma_fitting_regions(coords_neck_1, teffs_neck_1, direction='y', cutoff=cutoff, 
+                                                                        component=1, plot=False)
+    coords_fit_y2, teffs_fit_y2 = _isolate_sigma_fitting_regions(coords_neck_2, teffs_neck_2, direction='y', cutoff=cutoff,
+                                                                        component=2, plot=False)
+
+    sigma_y1, amplitude_y1, model_y1 = _fit_sigma_amplitude(coords_fit_y1, teffs_fit_y1, offset, direction='y', 
+                                                                        component=1, plot=False)
+    sigma_y2, amplitude_y2, model_y2 = _fit_sigma_amplitude(coords_fit_y2, teffs_fit_y2, offset, direction='y', 
+                                                                        component=2, plot=False)
+
+    new_teffs1 =  _compute_twoD_Gaussian(coords_neck_1, sigma_x1, sigma_y1, teffs_neck_1.max(), 
+                                                cutoff=cutoff, offset=offset, component=1)
+    new_teffs2 = _compute_twoD_Gaussian(coords_neck_2, sigma_x2, sigma_y2, teffs_neck_2.max(), 
+                                                cutoff=cutoff, offset=offset, component=2)
+
+    # print(cond1, len(cond1), len(new_teffs1))
+    # print(cond2, len(cond2), len(new_teffs2))
+    teffs1[cond1] = new_teffs1
+    teffs2[cond2] = new_teffs2
+
+    return teffs1, teffs2
+
+
+def lateral_transfer(t2s, teffs2, mixing_power, teff_ratio):
+    """
+    Scales the temperatures of the secondary to that of the primary only in a horizontal band the size of the contact's
+    neck. This implies mixing occurs due to mass transfer across the neck
+    """
+    x2s = t2s[:,0]
+    y2s = t2s[:,1]
+    z2s = t2s[:,2]
+
+    y2s_neck = y2s[x2s<1]
+    z2s_neck = z2s[x2s<1]
+    rs_neck = (y2s_neck**2+z2s_neck**2)**0.5
+    lat = np.min(rs_neck)
+    assert lat == np.min(z2s_neck)
+    filt = (z2s>-lat) & (z2s<lat)  # select band extending the (projected) height of the neck
+    c = (lat-np.abs(z2s[filt])) ** mixing_power
+    latitude_dependence = c / c.max()
+    teffs2[filt] *= 1 + (1 - teff_ratio) * latitude_dependence
+
+    return teffs2
+
+
+def isotropic_transfer(t2s, teffs2, mixing_power, teff_ratio):
+    """
+    Scales the temperatures of the secondary to that of the primary, parametrized by the radial distance from the
+    origin (which is the center of the primary)
+    """
+    d2s = np.sqrt(t2s[:,0]*t2s[:,0] + t2s[:,1]*t2s[:,1] + t2s[:,2]*t2s[:,2])
+    teffs2 *= 1 + (1-teff_ratio) * (1 - ((d2s - d2s.min()) / (d2s.max() - d2s.min()))) ** mixing_power
+    return teffs2
+
+
+def perfect_transfer(t2s, teff2s, teff_ratio):
+    """
+    Scales the temperatures of the secondary to that of the primary, implying perfect thermal mixing occurred deep in
+    the interior of the stars.
+    """
+    teff2s *= 1/teff_ratio
+    return teff2s
+
+
+def spotty_transfer(t2s, teffs2):
```

Comment:
I don't think this transfer mode will have any practical value because of the randomized spot positions.